lisa/sut_orchestrator/azure/common.py

```diff
@@ -243,13 +429,121 @@ def __hash__(self) -> int:
+            capabilities.append(SecurityProfileType.CVM)
+            capabilities.append(SecurityProfileType.Stateless)
+        self.security_profile = security_profile = search_space.SetSpace(
+            True, capabilities
+        )
+
+    def _parse_version(self, platform: "AzurePlatform") -> "SharedImageGallerySchema":
+        compute_client = get_compute_client(platform)
+        if not self.resource_group_name:
+            # /subscriptions/xxxx/resourceGroups/xxxx/providers/Microsoft.Compute/
+            # galleries/xxxx
+            rg_pattern = re.compile(r"resourceGroups/(.*)/providers", re.M)
+            galleries = compute_client.galleries.list()
+            for gallery in galleries:
+                if gallery.name and gallery.name.lower() == self.image_gallery:
+                    assert gallery.id, "'gallery.id' must not be 'None'"
+                    self.resource_group_name = get_matched_str(gallery.id, rg_pattern)
+                    break
+        if not self.resource_group_name:
+            raise LisaException(f"did not find matched gallery {self.image_gallery}")
+
+        if self.image_version.lower() == "latest":
+            gallery_images = (
+                compute_client.gallery_image_versions.list_by_gallery_image(
+                    resource_group_name=self.resource_group_name,
+                    gallery_name=self.image_gallery,
+                    gallery_image_name=self.image_definition,
+                )
+            )
+            time: Optional[datetime] = None
+            for image in gallery_images:
+                assert image, "'image' must not be 'None'"
+                assert image.name, "'image.name' must not be 'None'"
+                gallery_image = compute_client.gallery_image_versions.get(
+                    resource_group_name=self.resource_group_name,
+                    gallery_name=self.image_gallery,
+                    gallery_image_name=self.image_definition,
+                    gallery_image_version_name=image.name,
+                    expand="ReplicationStatus",
+                )
+                if not time:
+                    time = gallery_image.publishing_profile.published_date
+                    assert image, "'image' must not be 'None'"
+                    assert image.name, "'image.name' must not be 'None'"
+                    self.image_version = image.name
+                elif gallery_image.publishing_profile.published_date > time:
+                    time = gallery_image.publishing_profile.published_date
+                    assert image, "'image' must not be 'None'"
+                    assert image.name, "'image.name' must not be 'None'"
+                    self.image_version = image.name
+        return copy.copy(self)
+
 
 @dataclass_json()
 @dataclass
 class VhdSchema(AzureImageSchema):
     vhd_path: str = ""
     vmgs_path: Optional[str] = None
 
+    def _get_info(self, platform: "AzurePlatform") -> Optional[Dict[str, Any]]:
```

Comment:
Does it miss merge information from runbook?

Reply:
Previously VHD only copied hyperv generation from the runbook, but I implemented that logic in platform in a slightly different way. Now the hyperv generation provided in the runbook is the default value and it gets overridden if the image is tagged.